tests/test_roundtrip.py

```diff
@@ -214,3 +214,26 @@ def test_decimal_roundtrip(tmp_hyper, value, precision, scale, compat):
     pt.frame_to_hyper(tbl, tmp_hyper, table="test")
     result = pt.frame_from_hyper(tmp_hyper, table="test", return_type="pyarrow")
     compat.assert_frame_equal(result, tbl)
+
+
+def test_chunked_data_roundtrip(frame, tmp_hyper, compat):
+    if not isinstance(frame, pa.Table):
+        pytest.skip("only testing for pyarrow roundtrip")
+
+    pt.frame_to_hyper(
+        frame.to_reader(1),
+        tmp_hyper,
+        table="test",
+        process_params={"default_database_version": "4"},
+    )
+    result = pt.frame_from_hyper(tmp_hyper, table="test", return_type="pyarrow")
+    # A blind cast here may not be safe...but listing all expectations out
+    # would be extremely verbose
+    expected = frame.cast(result.schema)
+
+    # TODO: intervals are not working but that seems like a bug in nanoarrow instead
```

Comment:
This needs to be fixed upstream first in https://github.com/apache/arrow-nanoarrow/issues/754